compiler/rustc_mir_transform/src/pass_manager.rs

```diff
@@ -121,7 +121,10 @@ fn run_passes_inner<'tcx>(
                 validate_body(tcx, body, format!("before pass {name}"));
             }
 
-            tcx.sess.time(name, || pass.run_pass(tcx, body));
+            tcx.sess
+                .prof
+                .generic_activity_with_arg("mir_pass", name)
```

Comment:
Could you leave the pass name as activity name? `body.source.def_id()` can be used as arg.

Reply:
That should work, but they should gain a `mir_` prefix to fit with the profiling naming scheme.